examples/platform/efr32/uart.cpp

```diff
@@ -44,18 +44,27 @@ extern "C" {
 #define MIN(A, B) ((A) < (B) ? (A) : (B))
 #endif
 
+#ifdef EFR32MG24
+#define HELPER1(x) EUSART##x##_RX_IRQn
```

Comment:
what is "HELPER" ? Why is this not "EUSART_IRQ_NO" or something else that is clearer in wording and more readable?

below then we do `#define HELPER2(x) HELPER1(x)` ... why?

Reply:
Resolving because platform code... but personal opinion is that this is not good for maintenance.

Reply:
This code is copied from Silicon Labs' platform examples. 

> what is "HELPER" ? Why is this not "EUSART_IRQ_NO" or something else that is clearer in wording and more readable?

There's no good reason, that's just how it is. 

> below then we do #define HELPER2(x) HELPER1(x) ... why?

That's a limitation of gcc preprocessor. "If you want to stringify the result of expansion of a macro argument, you have to use two levels of macros." https://gcc.gnu.org/onlinedocs/gcc-4.8.5/cpp/Stringification.html